lib/core/versions/latest/Did.ts

```diff
@@ -29,12 +30,13 @@ export default class Did {
    * @param didMethodName The expected DID method given in the DID string. The method throws SidetreeError if mismatch.
    */
   private constructor (did: string, didMethodName: string) {
-    if (!did.startsWith(didMethodName)) {
+    this.didMethodName = didMethodName;
```

Comment:
```typescript
(did: string, public didMethodName: string) {
```

Reply:
Would do it if there were no other props or other props are declared and instantiated the same way. In this case prefer to keep the declaration of the `didMethodName` property explicit together with  other props for consistency.